src/mscorlib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs

```diff
@@ -203,6 +221,92 @@ public override void Return(T[] array, bool clearArray = false)
             }
         }
 
+        public bool Trim()
+        {
+            int milliseconds = Environment.TickCount;
+            MemoryPressure pressure = GetMemoryPressure();
+
+            ArrayPoolEventSource log = ArrayPoolEventSource.Log;
+            if (log.IsEnabled())
+                log.BufferTrimPoll(milliseconds, (int)pressure);
+
+            foreach (PerCoreLockedStacks bucket in _buckets)
+            {
+                bucket?.Trim((uint)milliseconds, Id, pressure, _bucketArraySizes);
+            }
+
+            if (pressure == MemoryPressure.High)
+            {
+                // Under high pressure, release all thread locals
```

Comment:
This is a little counter-intuitive to me.  The array stored in the TLS slot is going to be the most recently used one; presumably we're more interested in keeping that one around than ones in the per-core stacks.  But maybe this is the best we can do right now?

Reply:
Best for now.  We didn't want to risk a perf hit from GetTicks or code complexity at this point.